src/main/java/com/github/charlyb01/music_control/Utils.java

```diff
@@ -18,4 +21,20 @@ public static void print(final MinecraftClient client, final Text text) {
             }
         }
     }
+
+    public static float clamp(float num, float min, float max) {
+        return Math.min(Math.max(num, min), max);
+    }
+
+    public static float roundUp(float value, int places) {
+        BigDecimal bd = new BigDecimal(Float.toString(value));
+        bd = bd.setScale(places, RoundingMode.CEILING);
+        return bd.floatValue();
+    }
+
+    public static float roundDown(float value, int places) {
+        BigDecimal bd = new BigDecimal(Float.toString(value));
+        bd = bd.setScale(places, RoundingMode.FLOOR);
+        return bd.floatValue();
+    }
```

Comment:
(I thought I already wrote that but lost it)
Why would you use BigDecimal? All of these already exist for sure in Math (and/or Minecraft)

Reply:
On top of that (inefficiency), the use you make of them is useless. You should get rid of them

Reply:
I had many issues with this, for example it would give 20.000004 or 74.9999997, I could try using other methods but this is the one I know works

Reply:
It's just because they are float value, and floating numbers aren't real numbers (computer science). What you're doing does the exact same thing at a bigger cost. And the clamp is useless as the subtraction and the addition are not used in the same time. The min/max was already the perfect fit. You need to get rid of all of it

Reply:
I use clamp because if the volume is over 100%, it would snap to 100% but I'll remove it if you think it is unnecessary, what method do you suggest for the rounding? `Math.ceil(x / 100) * 100`?

Reply:
Fixed as of https://github.com/sf-inc/music_control/pull/20/commits/41084fdd8943996c328e49921fa0f40390b61bf0